### DISCUSSION TITLE
Global Data fetching

### DISCUSSION THREAD

**Discussion by santicros** [ElMassimo/iles#46] [Q&A] [answered]
Is it possible to fetch data for using in different places?

I was trying to do:
```js
// site.ts
import { $fetch } from "ohmyfetch";

const getData = async () => {
  const data = await $fetch("https://jsonplaceholder.typicode.com/todos/1");
  return data;
};

export default {
  title: "îles",
  description: "Islands of interactivity with Vue in Vite.js",
  test: getData(), // also tried `await getData()` but get an error that can't do top level await
};
```

But using `$site.test` on any template gets me an `[object Promise]`

**Comment by ElMassimo** [comment, marked as answer]
In your example, you can `const data = await site.test` in a `script setup` block in a Vue component.

Will add a recipes section to the docs with more info in the upcoming weeks.
